server/container/util-java8/src/main/java/org/apache/james/util/MissableOptional.java

```diff
@@ -0,0 +1,177 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.util;
+
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.apache.commons.lang.NotImplementedException;
+
+import com.google.common.base.Preconditions;
+
+public class MissableOptional<T> {
```

Comment:
Ok, trading is between genericity and expresivity. I will rename code. 

However **PatchedValue** is not a concept specific to JMAP. It can still benefit being in an other package... (It can be used, I don't know, but for instance for administration interface...)

Moreover, it is used by :
 - of course **server/protocols/jmap**
 - but also **server/data/data-jmap** (**VacationPatch**) where patch is generated from the response

You can not move it to server/protocols/jmap without creating circular dependency. It do not add anything to have it in data-jmap, IMO